libs/tracing/basic.ml

```diff
@@ -0,0 +1,4 @@
+let foo a b = a + b [@@trace "foo 1"]
```

Comment:
this looks more like tests, so maybe move in libs/tracing/tests/ (or in tests/tracing/)